src/vmm/src/builder.rs

```diff
@@ -244,15 +248,13 @@ fn create_vmm_and_vcpus(
         .map_err(Error::EventFd)
         .map_err(Internal)?;
 
+    let resource_manager = ResourceManager::new().map_err(ResourceAllocator)?;
+
     // Instantiate the MMIO device manager.
     // 'mmio_base' address has to be an address which is protected by the kernel
     // and is architectural specific.
-    let mmio_device_manager = MMIODeviceManager::new(
-        arch::MMIO_MEM_START,
-        arch::MMIO_MEM_SIZE,
-        (arch::IRQ_BASE, arch::IRQ_MAX),
-    )
-    .map_err(StartMicrovmError::RegisterMmioDevice)?;
+    let mmio_device_manager =
```

Comment:
Trying to figure out why we do not associate a device manager with a resource manager. Based on the type of device manager we are dealing with, we could call into the right AddressAllocator, right (I might be missing some aspects)

Reply:
This is what we do currently, though without a resource manager. Each device manager has an `AddressAllocator` and (maybe) an `IdAllocator`.

This works mostly fine, so far, but it doesn't scale:

When we add the ACPI devices they will need GSIs as well, so if we follow the same approach, create an `IdAllocator` for the "ACPI device manager", we would need to statically split the available GSIs between the two device types, introducing fragmentation. Having a single resource manager (per VM) avoids this. 

Also, not sure if it is important, but I was thinking that having a specific type that handles these resources, would allow us to more easily include that information to the snapshot state, so on restore we can avoid trying to re-allocate things in the exact same order. Btw, atm we do not have a way to allocate the exact same GSI, which is problematic